mcp_gateway/plugins/guardrails/lasso.py

```
@@ -171,14 +182,25 @@ def _extract_messages_from_request(
-        # If we couldn't find messages in the expected format, log a warning
+        # If no messages format, extract ALL string values from arguments
+        # This handles MCP tools like falcon-mcp that use filter/query args
+        if not messages and arguments:
+            all_strings: List[str] = []
+            self._extract_string_values(arguments, all_strings)
+            if all_strings:
+                # Combine all string values into a single user message
+                combined_content = " ".join(all_strings)
+                messages.append({"role": "user", "content": combined_content})
+                logger.info(
+                    f"Extracted {len(all_strings)} string values from arguments for Lasso check"
```

Ruff (0.14.13)

Logging statement uses f-string

(G004)